Ginger/GingerCoreCommon/VariablesLib/VariableSelectionList.cs

```diff
@@ -128,7 +131,36 @@ public override void ResetValue()
 
         public override void GenerateAutoValue()
         {
-            //NA
+            string trimString = "Options: ";
+            string[] listValues = Formula.Substring(trimString.Length).Split(',');
+            if (listValues.Length == 0)
+            {
+                Value = string.Empty;
+                return;
+            }
+
+            //Try to move to the next value if possible.
+            try
+            {
+                bool isLast = !OptionalValuesList.MoveNext();
+                Value = ((OptionalValue)OptionalValuesList.CurrentItem).Value;
+            }
+            catch (Exception e)
+            {
+                //If loop chechbox is disabled so return a proper message.
+                if (!IsLoopEnabled)
+                {
+                    Value = "Value is at the last in the list and no looping chechkbox is not enabled";
+                    Reporter.ToLog(eLogLevel.ERROR, "Value is at the last in the list and no looping chechkbox is not enabled");
```

Comment:
1- do not used exception catch has your if-else
2- check for all cases no loop/single value/no value in list
3- set clearer error message like "Auto value generating is not possible because only one value exist / np value exist / current value is last and looping is not allowed